### ISSUE TITLE
The label process during distillation

### ISSUE THREAD

**Issue by unlabeledData** [songmzhang/DSKD#30]
Thank you for your great work. I have a question about labels for loss computation.

According to the code `no_model_data["label"][i][:source_len-1] = -100` in `DistillDataset`, it seems DSKD only compute the output tokens loss for both SFT and Distill. Is my understanding correct? 

However, some repos still use input prompt for loss computation. For example, TRL's default dataloader `DataCollatorForLanguageModeling`. 
I want to know that is that (masking input prompt) better for distillation?

Sorry, I don't have enough LLM SFT experiences so I have the question.

**Comment by songmzhang** [comment]
> Thank you for your great work. I have a question about labels for loss computation.
> 
> According to the code `no_model_data["label"][i][:source_len-1] = -100` in `DistillDataset`, it seems DSKD only compute the output tokens loss for both SFT and Distill. Is my understanding correct?
> 
> However, some repos still use input prompt for loss computation. For example, TRL's default dataloader `DataCollatorForLanguageModeling`. I want to know that is that (masking input prompt) better for distillation?
> 
> Sorry, I don't have enough LLM SFT experiences so I have the question.

Yes, the standard loss computation of SFT is just calculating the loss on the output tokens. It is not about distillation, but the default setting for SFT( you can also see it in most fine-tuning repos like LLaMA-Factory, ms-swift, etc.). Some dataloaders may be designed for both pretraining and SFT, and thus didn't set the loss mask in the data collator for flexibility.

**Comment by unlabeledData** [comment]
Thank you for your nice reply. Very helpful~
